### PR TITLE
Cache more results

### PR DESCRIPTION
**Pull request by rjmac**
* Remove the notion of "cacheable query"
* Instead, store up to 10K of response body while streaming it
* If the response didn't overflow and the full body was sent (i.e., there was no exception during processing), stick it in the cache so it can be reused

### CODE REVIEW

File: soql-server-pg/src/main/scala/com/socrata/pg/server/analyzer2/ResultCache.scala [hunk 1]
```diff
@@ -2,13 +2,23 @@ package com.socrata.pg.server.analyzer2
...
 
 import com.socrata.http.server.util.{EntityTag, WeakEntityTag, StrongEntityTag}
-import com.socrata.soql.analyzer2._
+
+final abstract class ResultCache
 
 object ResultCache {
+  val log = LoggerFactory.getLogger(classOf[ResultCache])
+
+  private val uselessCache = new AtomicLong(0)
```
Comment (rlvoyer) [inline soql-server-pg/src/main/scala/com/socrata/pg/server/analyzer2/ResultCache.scala:20]: What's the `uselessCache` for?

Comment (rjmac) [reply soql-server-pg/src/main/scala/com/socrata/pg/server/analyzer2/ResultCache.scala]: Leftover; already deleted.

**Review by saraid — APPROVED** [review]
LGTM. I wonder if the limit should be configurable instead of hardcoded?

**Review by saraid — APPROVED** [review]
Not spotting any problems.

**Pull request merged**